walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,75 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.List;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<TimeRange> possibleTimes = new ArrayList<>();
+    List<TimeRange> conflictingTimes = new ArrayList<>();
+    Collection<String> attendees = request.getAttendees();
+
+    // Find every TimeRange that needs to be avoided
+    for (Event event : events) {
+      for (String attendee : event.getAttendees()) {
+        if (attendees.contains(attendee)) {
+          conflictingTimes.add(event.getWhen());
+          break;
+        }
+      }
+    }
```

Comment:
Generally we aim to keep methods short and simple. Since this method consists of several separate blocks of code, that each require a comment to explain what it does, one remedy is to try to extract one or more smaller, more focused, clearly-named private helper methods.